python/dgl/graphbolt/dataset.py

```diff
@@ -1,47 +1,113 @@
 """GraphBolt Dataset."""
 
+from typing import List
+
 from .feature_store import FeatureStore
 from .itemset import ItemSet, ItemSetDict
 
-__all__ = ["Dataset"]
+__all__ = ["Task", "Dataset"]
 
 
-class Dataset:
-    """An abstract dataset.
+class Task:
+    """A task.
 
-    Dataset provides abstraction for accessing the data required for training.
-    The data abstraction could be a native CPU memory block, a shared memory
-    block, a file handle of an opened file on disk, a service that provides
-    the API to access the data e.t.c. There are 3 primary components in the
-    dataset: *Train-Validation-Test Set*, *Feature Storage*, *Graph Topology*.
+    Task consists of several meta information and the *Train-Validation-Test Set*.
 
     *Train-Validation-Test Set*:
     The training-validation-testing (TVT) set which is used to train the neural
     networks. We calculate the embeddings based on their respective features
     and the graph structure, and then utilize the embeddings to optimize the
     neural network parameters.
+    """
 
-    *Feature Storage*:
-    A key-value store which stores node/edge/graph features.
+    def __init__(
+        self,
+        name: str,
+        num_classes: int,
+        num_labels: int,
+        train_set: ItemSet or ItemSetDict,
+        validation_set: ItemSet or ItemSetDict,
+        test_set: ItemSet or ItemSetDict,
+    ):
+        """Initialize a task.
+
+        Parameters
+        ----------
+        name : str
+            Task name.
+        num_classes : int
+            Number of classes.
+        num_labels : int
+            Number of labels.
+        train_set : ItemSet or ItemSetDict
+            Training set.
+        validation_set : ItemSet or ItemSetDict
+            Validation set.
+        test_set : ItemSet or ItemSetDict
+            Test set.
+        """
+        self._name = name
+        self._num_classes = num_classes
+        self._num_labels = num_labels
+        self._train_set = train_set
+        self._validation_set = validation_set
+        self._test_set = test_set
 
-    *Graph Topology*:
-    Graph topology is used by the subgraph sampling algorithm to
-    generate a subgraph.
-    """
+    @property
+    def name(self) -> str:
+        """Return the task name."""
+        return self._name
+
+    @property
+    def num_classes(self) -> int:
+        """Return the number of classes."""
+        return self._num_classes
+
+    @property
+    def num_labels(self) -> int:
+        """Return the number of labels."""
```

Comment:
Does this apply to all types of graph ML tasks?

Reply:
I think multi-labels applies to classification tasks, but not sure whether it applies to all kinds of classification tasks.